src/js/data-builder.js

```diff
@@ -199,14 +208,12 @@ const buildObjectNodeRelationship = (renderedNodes, filteredJson) => {
   });
 
   //Return nodes that are related to provided filtered json
-  return renderedNodes.filter((node) => relatedData.some((item) => item.name === node.name));
+  return renderedNodes.filter((node) => relatedData.some((item) => item.id === node.id));
 };
 
 const filterObjectsOnNodeClick = (json, nodes = []) => {
   let filteredJson = json.filter((element) => {
-    return nodes.filter((node) => node === element.GRANTED_TO_NAME && element.GRANTED_TO_TYPE === 'ROLE').length > 0
-      ? true
-      : false;
+    return nodes.filter((node) => node === element.toId).length > 0 ? true : false;
```

Comment:
this ternary is redundant. The predicate will return the same values and the ternary makes this code harder to read

Reply:
How is this redundant? if no node found, it return false. 

It's the opposite, for simple if else statements ternary operators are preferred. in fact react uses this heavily.